### ISSUE TITLE
All hosts blocked if empty line in host blocklist

### ISSUE THREAD

**Issue by toofar** [qutebrowser/qutebrowser#4325]
hseg reported this on IRC.

**Version info (see `:version`)**:
git master
CPython: 3.6.5
Qt: 5.11.1
PyQt: 5.11.2

I also reproduced it at work with a recent windows release, v1.4.2 I think. 

**Does the bug happen if you start with `--temp-basedir`?** (if applicable):
Yes.

**Description**
If you put a blank line in `blocked-hosts` in the config folder all hosts will be blocked. Including `qute://` URLs and the web inspector ones `chrome-devtools://`. Why? Use `--debug --logfilter webview --debug-flag log-requests` and observe the `blob:` URLs. In the request interceptor they have a ~first party URL~ host part of `''` so they match the entry that gets parsed from the blank line!

That makes some kind of sense but raises some questions. Why do these blob requests being blocked block the main page load? Where does the UUID in the blob URL come from? Where do these requests even come from? They are not from the page, try it with a site like `http://icanhazip.com/` which doesn't do much at all.

**How to reproduce**

Put a blank line in `blocked-hosts`, you don't need anything else in there but you can. Either restart or use `:ablock-update`. (Re)Load a page.
I assume this would also be the case with downloaded blocklists but haven't tested.

I also tried with (also git master):
CPython: 3.5.3
Qt: 5.7.1
PyQt: 5.7

and there were none of the offending blob URLs and all hosts were not blocked. I haven't got a 5.10 or 5.9 to test with.

**Comment by The-Compiler** [comment]
That's odd. Note that they don't get blocked because of the empty `firstPartyUrl()` (`is_blocked()` checks `url.host()`, not `first_party_url`) - it's because e.g. `QUrl('blob:a88814b4-df47-44cf-8c62-54062d071e8d').host() == ''`.

Here are the requests:

```
11:01:12 DEBUG    webview    interceptor:interceptRequest:70 GET http://icanhazip.com/, first-party http://icanhazip.com/, resource ResourceTypeMainFrame, navigation NavigationTypeTyped
11:01:12 DEBUG    webview    browsertab:_set_load_status:817 load status for <qutebrowser.browser.webengine.webenginetab.WebEngineTab tab_id=0 url='http://icanhazip.com/'>: LoadStatus.loading
11:01:12 DEBUG    webview    interceptor:interceptRequest:70 GET blob:http://icanhazip.com/b966a30c-032e-442f-8a73-547552296066, first-party , resource ResourceTypeMainFrame, navigation NavigationTypeTyped
11:01:13 DEBUG    webview    browsertab:_set_load_status:817 load status for <qutebrowser.browser.webengine.webenginetab.WebEngineTab tab_id=0 url='http://icanhazip.com/'>: LoadStatus.success
```

I don't see this with Qt 5.10, and I don't see it with Qt 5.12 either, so maybe this just was some kind of bug with Qt 5.11...

**Comment by toofar** [comment]
This answer https://stackoverflow.com/a/30881444 says that those blob URLs are used to refer to something in renderer memory, like if you load an image via XHR and don't want to shove it in a data URL you can ask the browser to generate such a URL. This problem definitely happens with javascript disabled though. They are always in this case of the form `blob:{first-party-url}/{UUID}` so we could check for that format and allow them. But then other URLs without a host part would have the same issue, there is no way to block them with a host blocker. You can now whitelist them though!

**Comment by The-Compiler** [comment]
I think we should:

- Ignore empty lines when reading adblock files (no point in interpreting them as hosts)
- Either ignore all URLs without a host when figuring out whether to block it,
- or at least ignore all `blob:` URLs.

**Comment by gebulmer** [comment]
> Ignore empty lines when reading adblock files (no point in interpreting them as hosts)

This can be done alongside #4473 and could be marked as 'easy'

**Comment by arza-zara** [comment]
I assume this was fixed by #4863.
